docs/pages/application-access/cloud-apis/google-cloud.mdx

```diff
@@ -0,0 +1,615 @@
+```
+
+When a user with the `google-cloud-cli-access` role authenticates to a Google
+Cloud CLI via Teleport, the Teleport Auth Service populates the
+`{{internal.gcp_service_accounts}}` template variable with any Google Cloud
+service accounts you have assigned to the user.
+
+Assign the `teleport-vm-lister` service account we created earlier (or another
+service account) to your Teleport user by running the following command:
+
+```code
+$ tctl users update <Var name="teleport-user" /> \
+--set-gcp-service-accounts teleport-vm-lister@${PROJECT_ID?}.iam.gserviceaccount.com
+```
+
+This command uses the `--set-gcp-service-accounts` flag to add Google Cloud
+service accounts to a user. You can assign `--set-gcp-service-accounts` to
+multiple service account URIs, separated by commas.
+
+Create the role:
+
+```code
+$ tctl create -f google-cloud-cli-access.yaml
+```
+
+</TabItem>
+<TabItem options="Dynamic Identities" label="SAML/OIDC Connectors">
+
+In your identity provider, define a custom SAML attribute or OIDC claim called
+`gcp_service_accounts`. Each user's `gcp_service_accounts` attribute or claim
+must be a list of Google Cloud service account URIs, using the following format:
+
+```text
+/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/RESOURCE_GROUP_NAME/providers/Microsoft.ManagedIdentity/userAssignedIdentities/IDENTITY_NAME
+```
+
+For example, you can set a user's `gcp_service_accounts` to `teleport-vm-lister`
+by using the following URI, replacing `my-project` with the name of your Google
+Cloud project:
+
+```text
+teleport-vm-lister@my-project.iam.gserviceaccount.com
+```
+
+Create a file called `google-cloud-cli-access.yaml` with the following content:
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: google-cloud-cli-access
+spec:
+  allow:
+    app_labels:
+      '*': '*'
+    gcp_service_accounts:
+      - '{{external.gcp_service_accounts}}'
+```
+
+When a user with the `google-cloud-cli-access` role authenticates to a Google
+Cloud CLI via Teleport, the Teleport Auth Service populates the
+`{{external.gcp_service_accounts}}` template variable with any Google Cloud
+service accounts you have assigned to the user.
+
+Create the role:
+
+```code
+$ tctl create -f google-cloud-cli-access
+```
+
+</TabItem>
+<TabItem options="Static Identities" label="All Authentication Methods">
+
+Define a role with access to specific Google Cloud service accounts, which means
+that Teleport users who assume this role can use those (and only those)
+identities to execute commands via a Google Cloud CLI.
+
+Create a file called `google-cloud-cli-access.yaml` with the following content,
+replacing `my-project` with the ID of your Google Cloud project in the value of
+`gcp_service_accounts`:
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: google-cloud-cli-access
+spec:
+  allow:
+    app_labels:
+      '*': '*'
+    gcp_service_accounts:
+      - teleport-vm-lister@my-project.iam.gserviceaccount.com
```

Comment:
```suggestion
      - teleport-vm-lister@${PROJECT_ID?}.iam.gserviceaccount.com
```

Reply:
Since this is part of a YAML snippet, I'm not sure a shell variable would make sense here. Should I make `my-project` more prominent as a placeholder value by capitalizing it?

Reply:
Ah, I thought YAML snippets are treated the same way as the shell snippets. I guess capitalization works, too, anything really to make sure this gets updated.